### PR TITLE
Update to scala 2.12

### PR DESCRIPTION
**Pull request by er1c**
There are a few failing tests that I'm not sure about - especially related to comments?

Since I spent a half day hacking on this, I figured I'd create a PR of my progress.

### CODE REVIEW

**Comment by nightscape** [comment]
Hi @er1c, thanks a lot for this PR!
scalagen is in dire need for a little love :wink:
I just added Github actions to run the tests in an "official" CI system.
The tests are failing even without your changes.
Could you check if the failures are the same?
https://github.com/scalagen/scalagen/runs/889953500?check_suite_focus=true

**Comment by er1c** [comment]
@nightscape thanks I'll take a look or likely just repush the branch so it gets ran

I started hacking on the jp3 branch a bit too - which is daunting considering all of different type changes

File: scalagen/src/main/scala/com/mysema/scalagen/Annotations.scala [hunk 1]
```diff
@@ -34,7 +36,7 @@ class Annotations(targetVersion: ScalaVersion) extends UnitTransformerBase {
     
   override def visit(n: AnnotationDecl, arg: CompilationUnit) = {
     // turns annotations into StaticAnnotation subclasses
-    if (targetVersion >= Scala210) {
+    if (targetVersion.sortOrder >= Scala210.sortOrder) {
```
Comment (er1c) [inline scalagen/src/main/scala/com/mysema/scalagen/Annotations.scala]: This bugged the shit out of me for an hour+,  I can't figure out why the `Ordered` isn't  working here

File: scalagen/src/main/scala/com/mysema/scalagen/ScalaVersion.scala [hunk 2]
```diff
@@ -3,7 +3,7 @@ package com.mysema.scalagen
 import scala.util.{Properties => ScalaProperties}
 
 sealed abstract class ScalaVersion(val sortOrder: Int) extends Ordered[ScalaVersion] {
-  def compare(that: ScalaVersion): Int = sortOrder.compare(that.sortOrder)
+  override final def compare(that: ScalaVersion): Int = sortOrder.compare(that.sortOrder)
```
Comment (er1c) [inline scalagen/src/main/scala/com/mysema/scalagen/ScalaVersion.scala:6]: added to  make sure it  was  implementing the right signature  :/

**Comment by er1c** [comment]
I rebased and cleaned it up a little, hopefully CI matches master

**Comment by er1c** [comment]
```
[error] Failed: Total 149, Failed 5, Errors 0, Passed 144
[error] Failed tests:
[error] 	com.mysema.scalagen.SerializationTest
[error] 	com.mysema.scalagen.ScalaCompilationTest
[error] (scalagen/test:test) sbt.TestsFailedException: Tests unsuccessful
```

The failures match master

File: scalagen/src/main/scala/com/mysema/scalagen/Annotations.scala (hunk 1 above)
Comment (nightscape) [reply scalagen/src/main/scala/com/mysema/scalagen/Annotations.scala]: If this is only for compatibility with Scala < 2.10 then we can get rid of it alltogether.

**Comment by nightscape** [comment]
If the PR doesn't introduce additional errors then I'm fine with merging it as is.
It's unfortunate that there are errors even on master, but I don't have time to fix them atm and letting PRs hang around for too long is the slow death of a Github project :open_mouth:

**Comment by er1c** [comment]
@nightscape what version(s) of scala do you want to support? `2.12`?

**Comment by nightscape** [comment]
Yes, I would say 2.12 and later.

**Comment by er1c** [comment]
@nightscape this  should be ready

```
error] Failed: Total 149, Failed 5, Errors 0, Passed 144
[error] Failed tests:
[error] 	com.mysema.scalagen.SerializationTest
[error] 	com.mysema.scalagen.ScalaCompilationTest
```

**Comment by nightscape** [comment]
@er1c thanks a lot!

**Pull request merged**